### ISSUE TITLE
[repo] flatten i18n prop objects

### ISSUE THREAD

**Issue by tay1orjones** [carbon-design-system/carbon-addons-iot-react#1811]
### What package is this for?

- [x] React
- [ ] Angular

### Summary

Would like to propose that we flatten all i18n prop objects out to individual properties no longer within an object.

- [ ] All translatable strings should be standardized to be suffixed with `Text`. If an i18n prop does not have `Text` as it's suffix, we add it.
- [ ] Write and provide a codemod script to help consumers migrate from the old props to the new ones (stretch goal)

```diff
const propTypes = {
-  i18n: PropTypes.shape({
-    openEditorButton: PropTypes.string,
-    cardTitle: PropTypes.string,
-    description: PropTypes.string,
-  }),
+  openEditorButtonText: PropTypes.string,
+  cardTitleText: PropTypes.string,
+  descriptionText: PropTypes.string,
};
```

### Why?

It adds additional complexity to the codebase with no quantifiable value. Currently we have to merge defaultProps.i18n with passed i18n props to ensure defaults and passed parameters are all together.

https://github.com/carbon-design-system/carbon-addons-iot-react/blob/a6444bd4ee64aa7bb4c551784175f8adebdb39ea/src/components/CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItem.jsx#L142

It's also something that we do differently than Carbon and causes our API to be inconsistent between components we modify from Carbon and ones that we re-export with no modifications.

### Specific timeline issues / requests

This would be a breaking change and could be included with v3

**Comment by bjornalm** [comment]
I think the value of the existing i18n object lies in the readability of having the props grouped. As developer I don't necessarily know the name of the i18n prop I'm after so I scan the i18n object to quickly find it. When a component has 5-6 props it doesn't matter, but for larger components with a lot of props it becomes more difficult to find the i18n prop that fits my need. Still, this effort is probably negligible so I agree that it would be best if our API was consistent on this point.

**Comment by davidicus** [comment]
Further discussion required. A few points about flattening object:
- **dev experience**: as @bjornalm mentions above. 
- **readability**:  For some of our composed components the list of strings we must pass along get to be pretty long. 
- Perhaps we consistently diverge. The change would be to make sure all strings are passed in an `i18n` object and we provide helper function or eslint plugin to enforce

**Comment by kevinsperrine** [comment]
I vote for moving everything into the i18n prop. It diverges from carbon, yes, but I think it's a much more readable and easily understandable change.

**Comment by kevinsperrine** [comment]
closed via #3110
